ext/stackprof/stackprof.c

```diff
@@ -187,16 +188,24 @@ frame_i(st_data_t key, st_data_t val, st_data_t arg)
 
     rb_hash_aset(results, rb_obj_id(frame), details);
 
-    name = rb_profile_frame_full_label(frame);
-    rb_hash_aset(details, sym_name, name);
+    if (frame == _stackprof.fake_gc_frame) {
+	name = rb_str_new_literal("(garbage collection)");
+	file = rb_str_new_literal("");
```

Comment:
Might be worth it to create these two strings once upfront and mark them as globals to re-use as well.